Src/FluentAssertions/Primitives/StringEqualityValidator.cs

```diff
@@ -27,14 +27,56 @@ protected override bool ValidateAgainstSuperfluousWhitespace()
 
         protected override bool ValidateAgainstLengthDifferences()
         {
+            // Logic is a little bit convoluted because I want to avoid calculation
+            // of mismatch segment in case of equalLength == true for performance reason.
+            // If lazy version of FailWith would be introduced, calculation of mismatch
+            // segment can be moved directly to FailWith's argument
+            bool equalLength = subject.Length == expected.Length;
+
+            string mismatchSegment = GetMismatchSegmentForStringsOfDifferentLengths(equalLength);
+
             return assertion
-                .ForCondition(subject.Length == expected.Length)
+                .ForCondition(equalLength)
                 .FailWith(
-                    ExpectationDescription + "{0} with a length of {1}{reason}, but {2} has a length of {3}.",
+                    ExpectationDescription + "{0} with a length of {1}{reason}, but {2} has a length of {3}, differs near " + mismatchSegment +  ".",
                     expected, expected.Length, subject, subject.Length)
                 .SourceSucceeded;
         }
 
+        private string GetMismatchSegmentForStringsOfDifferentLengths(bool equalLength)
+        {
+            if (equalLength)
+            {
+                return "";
+            }
+
+            int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparisonMode);
+
+            // If there is no difference it means that either
+            // * subject starts with expected or
+            // * expected starts with subject
+            if (indexOfMismatch == -1)
+            {
+                // If subject is shorter we are sure that expected starts with subject
+                if (subject.Length < expected.Length)
+                {
+                    // Subject is shorter so we point at its last character.
+                    // We would like to point at next character as it is the real
+                    // index of first mismatch, but we need to point at character existing in
+                    // subject, so the next best thing is the last subject character.
+                    indexOfMismatch = Math.Max(0, subject.Length - 1);
+                }
+                else
+                {
+                    // If subject is longer we are sure that subject starts with expected
+                    // and we point at first character after expected.
+                    indexOfMismatch = Math.Max(0, expected.Length);
```

Comment:
🔧 `Math.Max` is not be needed as `expected.Lengh` is `>= 0`.

Reply:
Good catch, thanks.